### DISCUSSION TITLE
Part name

### DISCUSSION THREAD

**Discussion by vaggarath** [google/model-viewer#3138] [Q&A] [answered]
I'm working with a bunch a 3D models in GLTG/GLB format. Each of these models has several parts. 

When the viewer is initializing i retrieve each parts to build a couple of buttons associated with each (show / hide). Since there is sometimes lots of parts i'm displaying the part name along-side those buttons for the user to know which part he can hide / display

The thing is i'm not retrieving the name i thought i'd get. 

Considering the image within this message, i thought i'd get the "main node" name (jaw_down). Instead i'm getting the last child, of the said node, name (the name of the material).
Thus, I tried to search but i can't find the part "true" name but only the one of the material. 

More than juste information, i'm asking because if we can anticipate the naming for futur models, we're going to have lots of work for the already made ones. If there is no way to retrieve this name it's an hassle but not the end of the world ;)

The way i retrieve the parts (maybe it's there i'm not doing things right)

```
const modelViewer = document.getElementById("animatediewer");
const parts = modelViewer.model.materials;
const parts = modelViewer.model.materials; //returns an array

for(let i = 0; i < parts.length; i++){
   /* some code to generate the buttons*/
   divName.innerText = parts[i].name
}
```
I tried ton console.log the parts to check the data they got, but this "true"name is nowhere to be found

Thanks for any help. Sorry is this issue is deemed useless ^^'

![image](https://user-images.githubusercontent.com/44065516/150014844-01140238-e470-4682-8df0-616f378453a3.png)

**Comment by elalish** [comment, marked as answer]
yes, the materials are indeed the glTF materials, which are not parts/meshes/primitives. Our API intentionally mirrors glTF (which doesn't have anything called a part). We also intentionally don't expose the glTF meshes/primitives in our API for simplicity. So yes, you'll need to  make sure the materials themselves are named in the glTF file, which might require looking at the  exporter. You might  find  it helpful to open your file in our [editor](https://modelviewer.dev/editor/); the last tab  is the glTF JSON, where you can directly find the names and  such.

**Reply by vaggarath** [reply]
Thanks for your answer.
I'm still lacking about 3D modelling so i wanted to  be sure!
Thank you again for taking time to explain.
